core/arch/x86_64/kernel/link.mk

```diff
@@ -17,6 +17,7 @@ link-ldflags += -T $(link-script-pp) -Map=$(link-out-dir)/tee.map
 link-ldflags += --sort-section=alignment
 link-ldflags += --fatal-warnings
 link-ldflags += --gc-sections
+link-ldflags += -z noseparate-code # Align with ARM, do not seperate PT_LOAD segment
```

Comment:
Why does Arm not use this link config?

Reply:
From ld v > 2.31, x86 will separate code by default, arm does not.